scipy/ndimage/_multimethods.py

```diff
@@ -0,0 +1,954 @@
+                   mode="reflect", cval=0.0, origin=0):
+    return input, footprint, mark_non_coercible(output)
+
+
+@create_ndimage(_input_size_footprint_output_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def median_filter(input, size=None, footprint=None, output=None,
+                  mode="reflect", cval=0.0, origin=0):
+    return input, footprint, mark_non_coercible(output)
+
+
+def _input_arg2_size_footprint_output_replacer(args, kwargs, dispatchables):
+    """
+    uarray argument replacer for signatures involving required ``input``
+    and a non-dispatchable second argument with ``size``, ``footprint``
+    and ``output`` kwarg.
+    """
+    def self_method(input, arg2, size=None, footprint=None,
+                    output=None, *args, **kwargs):
+        return (dispatchables[0], arg2, size, dispatchables[1],
+                dispatchables[2]) + args, kwargs
+
+    return self_method(*args, **kwargs)
+
+
+@create_ndimage(_input_arg2_size_footprint_output_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def rank_filter(input, rank, size=None, footprint=None, output=None,
+                mode="reflect", cval=0.0, origin=0):
+    return input, footprint, mark_non_coercible(output)
+
+
+@create_ndimage(_input_arg2_size_footprint_output_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def percentile_filter(input, percentile, size=None, footprint=None,
+                      output=None, mode="reflect", cval=0.0, origin=0):
+    return input, footprint, mark_non_coercible(output)
+
+
+def _generic_filter1d_arg_replacer(args, kwargs, dispatchables):
+    """
+    uarray argument replacer for generic_filter1d
+    """
+    def self_method(input, function, filter_size, axis=-1,
+                    output=None, *args, **kwargs):
+        return (dispatchables[0], arg2, arg3,
+                axis, dispatchables[1]) + args, kwargs
+
+    return self_method(*args, **kwargs)
+
+
+@create_ndimage(_generic_filter1d_arg_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def generic_filter1d(input, function, filter_size, axis=-1,
+                     output=None, mode="reflect", cval=0.0, origin=0,
+                     extra_arguments=(), extra_keywords=None):
+    return input, mark_non_coercible(output)
+
+
+@create_ndimage(_input_arg2_size_footprint_output_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def generic_filter(input, function, size=None, footprint=None,
+                   output=None, mode="reflect", cval=0.0, origin=0,
+                   extra_arguments=(), extra_keywords=None):
+    return input, footprint, mark_non_coercible(output)
+
+
+############################################
+""" fourier multimethods """
+############################################
+
+def _fourier_multimethods_arg_replacer(args, kwargs, dispatchables):
+    """
+    uarray argument replacer for fourier multimethods.
+    """
+    def self_method(input, arg2, n=-1, axis=-1, output=None, *args, **kwargs):
```

Comment:
> `arg2`

This doesn't work, the names need to match in order for keyword-arguments to work correctly.

Reply:
Thanks, Peter for catching this. Sorry I missed this simple thing when making the changes for `arg` inputs issue.